pkg/scanner/scan.go

```diff
@@ -110,6 +110,11 @@ func (s Scanner) ScanArtifact(ctx context.Context, options types.ScanOptions) (t
 	if err != nil {
 		return types.Report{}, xerrors.Errorf("failed analysis: %w", err)
 	}
+	defer func() {
+		if err := s.artifact.Clean(artifactInfo); err != nil {
+			log.Logger.Warnf("The artifact %q wasn't removed: %v", artifactInfo.Name, err)
```

Comment:
We may not remove the artifact.

```suggestion
			log.Logger.Warnf("Failed to clean the artifact %q: %v", artifactInfo.Name, err)
```

Reply:
updated